### PR TITLE
Polish 7.6 Release Notes

### PR DESCRIPTION
**Pull request by nathan-contino**
Polish for 7.6, pre-release

### Contributor Checklist
- [X] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [X] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [X] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [X] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [X] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [X] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [X] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [X] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [X] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -82,96 +82,102 @@ vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv -->
...
-In previous versions after a compilation failure the next compilation was not incremental but a full recompilation instead.
-
-With this version, Java and Groovy incremental compilation will work incrementally also after a failure.
-
-This improves experience with compilation when working iteratively on some Java or Groovy code, e.g. when iteratively running compile or test tasks from an IDE.
+Gradle supports [Java incremental compilation](userguide/java_plugin.html#sec:incremental_compile)
+by default and [Groovy incremental compilation](userguide/groovy_plugin.html#sec:incremental_groovy_compilation)
+as an opt-in experimental feature.
 
+In previous versions, a compilation failure caused the next compilation to fully recompile.
```
Comment (akshayabd) [inline subprojects/docs/src/docs/release/notes.md:89]: Hmmm maybe “to perform a full recompile” is a better ending?

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -82,96 +82,102 @@ vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv -->
...
+In Gradle 7.6 and above, Java and Groovy incremental compilation works even after a failure.
```
Comment (akshayabd) [inline subprojects/docs/src/docs/release/notes.md:90]: Hmm is it more correct to say “From Gradle 7.6 and above”?

**Review by hythloda — COMMENTED** [review]
I really like the added "For more information" pattern.  We need to make sure that is consistent for the ones on the bottom too.

**Review by akshayabd — COMMENTED** [review]
Changes look really good - just a couple of minor questions/comments

File: subprojects/docs/src/docs/release/notes.md (hunk 2 above)
Comment (akshayabd) [reply subprojects/docs/src/docs/release/notes.md:90]: Or just “From Gradle 7.6”

Comment (nathan-contino) [reply subprojects/docs/src/docs/release/notes.md:90]: "Starting in Gradle 7.6" is the better pattern I've seen -- I'll switch to that.

File: subprojects/docs/src/docs/release/notes.md [hunk 3]
```diff
@@ -82,96 +82,103 @@ vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv -->
...
-Gradle already supports [Java incremental compilation](userguide/java_plugin.html#sec:incremental_compile) by default and [Groovy incremental compilation](userguide/groovy_plugin.html#sec:incremental_groovy_compilation) as an opt-in experimental feature.
-
-In previous versions after a compilation failure the next compilation was not incremental but a full recompilation instead.
-
-With this version, Java and Groovy incremental compilation will work incrementally also after a failure.
-
-This improves experience with compilation when working iteratively on some Java or Groovy code, e.g. when iteratively running compile or test tasks from an IDE.
+Gradle supports [Java incremental compilation](userguide/java_plugin.html#sec:incremental_compile)
+by default and [Groovy incremental compilation](userguide/groovy_plugin.html#sec:incremental_groovy_compilation)
+as an opt-in experimental feature.
```
Comment (alllex) [inline subprojects/docs/src/docs/release/notes.md:87]: @asodja This should probably at the end be synchronized with your changes in https://github.com/gradle/gradle/pull/21878

Comment (asodja) [reply subprojects/docs/src/docs/release/notes.md:87]: Thanks for pointing that out. I believe so yes.

**Comment by nathan-contino** [comment]
@bot-gradle test and merge

**Comment by nathan-contino** [comment]
@bot-gradle test and merge

File: subprojects/docs/src/docs/release/notes.md [hunk 4]
```diff
@@ -254,37 +270,37 @@ Related issues:
 <a name="jvm"></a>
 ### JVM
 
-#### Added Java 19 Support
+#### Added Support for Java 19
 
 Gradle 7.6 supports compiling, testing and running on Java 19.
 
-#### Introduced strongly-typed `dependencies` block for JVM test suites
+#### Introduced named dependency arguments in the Kotlin DSL for external dependencies
```
Comment (tresat) [inline subprojects/docs/src/docs/release/notes.md:277]: I think this is the least interesting and should go at the bottom of the JVM section.

**Pull request merged**